src/lib.rs

```diff
@@ -76,28 +86,372 @@ impl ULinkZenoh {
             .replace("//", "\\/");
         Ok(zenoh_key)
     }
+
+    // TODO: We need a standard way in uprotocol-rust to change UUID to String
+    fn uuid_to_string(uuid: &Uuid) -> String {
+        format!("{}:{}", uuid.msb, uuid.lsb)
+    }
+
+    async fn send_publish(
+        &self,
+        zenoh_key: &str,
+        payload: UPayload,
+        attributes: UAttributes,
+    ) -> Result<(), UStatus> {
+        // Get the data from UPayload
+        let Some(Data::Value(buf)) = payload.data else {
+            // TODO: Assume we only have Value here, no reference for shared memory
+            return Err(UStatus::fail_with_code(
+                UCode::InvalidArgument,
+                "Invalid data",
+            ));
+        };
+
+        // Serialized UAttributes into protobuf
+        // TODO: Should we map priority into Zenoh priority?
```

Comment:
Yes

Reply:
The problem is how to map it:
Perhaps I can provide my initial idea:
Zenoh priority: https://zenoh-python.readthedocs.io/en/0.10.1-rc/#zenoh.Priority
uProtocol priority: https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/qos.adoc

CS0: BACKGROUND
CS1: DATA_LOW
CS2: DATA
CS3: DATA_HIGH
CS4: INTERACTIVE_LOW
CS5: INTERACTIVE_HIGH
CS6: REAL_TIME

Do you think this makes sense?

Reply:
Done.